hy/compiler.py

```diff
@@ -2035,7 +2033,12 @@ def compile_return(self, expr):
     def compile_class_expression(self, expressions):
         def rewire_init(expr):
             new_args = []
-            if expr[0] == HySymbol("setv"):
+            if not isinstance(expr, HyExpression):
+                raise HyTypeError(expr,
+                                  "`defclass` expects expressions, received `{}`".format(
```

Comment:
Technically, things like `42` are also expressions, so this message is confusing. It should say "HyExpressions" if that's what you actually want. But I'm not sure why we'd want to reject such things. Python doesn't.

Reply:
No, that's a very good point. But that, I think, makes this PR simpler.